test/utils/skupper/cli/tester.go

```diff
@@ -81,6 +108,44 @@ func RunScenarios(t *testing.T, scenarios []TestScenario) {
 	}
 }
 
+// Runs a list of []TestScenario in parallel.  Each scenario will run in
+// parallel to the other.  However, the RunScenariosParallel call will only
+// return when all of them finish.  This allows the caller function to do
+// other in-test steps that depend on the RunScenariosParallel items finishing.
+//
+// To implement and signify that in the output, the steps are enclosed in a
+// test called simply 'parallel'.
+func RunScenariosParallel(t *testing.T, scenarios []TestScenario) {
```

Comment:
Instead of duplicating what RunScenarios is doing here, why not doing:

```
func RunScenarios(t *testing.T, scenarios []TestScenario) {
	runScenarios(t, scenarios, false)
}

func RunScenariosParallel(t *testing.T, scenarios []TestScenario) {
	runScenarios(t, scenarios, true)
}
func runScenarios(t *testing.T, scenarios []TestScenario, parallel bool) {
    ...
    ...
    ...
				if parallel && base.ShouldRunScenariosInParallel() {
    ...
    ...
    ...
```

Reply:
What if in case parallel is true, you add the extra "parallel" test level? Otherwise, you just run as is.
Or eventually you prefix the scenario name.